src/frontend/sharedComponents/GPSPill.tsx

```diff
@@ -1,33 +1,19 @@
-import React, {FC} from 'react';
 import {StyleSheet, TouchableOpacity} from 'react-native';
+import {UIActivityIndicator} from 'react-native-indicators';
 import {DARK_GREY, WHITE} from '../lib/styles';
-import {GpsErrorIcon, GpsSearchingIcon, GpsGoodIcon} from './icons';
-import {useSharedLocationContext} from '../contexts/SharedLocationContext';
-import {getLocationStatus} from '../lib/utils';
+import {type LocationStatusResult} from '../lib/utils';
+import {GpsErrorIcon, GpsGoodIcon, GpsSearchingIcon} from './icons';
 import {BodyText} from './Text/BodyText';
-import {UIActivityIndicator} from 'react-native-indicators';
-import type {LocationProviderStatus} from 'expo-location';
 
-type GPSPillProps = {
+type GPSPillProps = LocationStatusResult & {
   onPress?: () => void;
-  locationProviderStatus?: LocationProviderStatus;
 };
 
-export const GPSPill: FC<GPSPillProps> = ({
-  onPress,
-  locationProviderStatus,
-}) => {
-  const {locationState, fgPermissions} = useSharedLocationContext();
```

Comment:
i identified this usage of the context as unnecessary and refactored this component to take the relevant information as props, which makes testing the component much more straightforward.